src/NServiceBus.Core/Serializers/MessageDeserializerResolver.cs

```diff
@@ -1,15 +1,24 @@
 ﻿namespace NServiceBus
 {
+    using System;
     using System.Collections.Generic;
-    using System.Linq;
     using Serialization;
 
     class MessageDeserializerResolver
     {
         public MessageDeserializerResolver(IMessageSerializer defaultSerializer, IEnumerable<IMessageSerializer> additionalDeserializers)
         {
             this.defaultSerializer = defaultSerializer;
-            serializersMap = additionalDeserializers.ToDictionary(key => key.ContentType, value => value);
+
+            foreach (var additionalDeserializer in additionalDeserializers)
+            {
+                if (serializersMap.ContainsKey(additionalDeserializer.ContentType))
+                {
+                    throw new Exception($"Multiple deserializers are registered for content-type '{additionalDeserializer.ContentType}'. Remove ambiguous deserializers.");
```

Comment:
How about we list the `IMessageSerializer` type that conflicts, and tell the user to remove that one, maybe even show them how to do it ?
Example:
```
The following deserializers handle the same 'contenttype' content-type:
- '<IMessageSerializer type1>' 
- '<IMessageSerializer type2>' 
NServiceBus does not know which one to pick for deserializing messages with 'contenttype' content-type.
To remove this ambiguity, you need to de-register one of the listed deserializers.
```